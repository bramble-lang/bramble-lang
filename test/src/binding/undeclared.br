fn my_main() -> i64 {
    root::std::io::writei64ln(x);
    return;
}