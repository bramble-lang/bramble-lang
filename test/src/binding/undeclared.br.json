[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 33], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [33, 35], "ok": "Operator"},
{"id": 10, "stage": "lexer", "source": [35, 38], "ok": "Identifier"},
{"id": 11, "stage": "lexer", "source": [38, 40], "ok": "Operator"},
{"id": 12, "stage": "lexer", "source": [40, 42], "ok": "Identifier"},
{"id": 13, "stage": "lexer", "source": [42, 44], "ok": "Operator"},
{"id": 14, "stage": "lexer", "source": [44, 54], "ok": "Identifier"},
{"id": 15, "stage": "lexer", "source": [54, 55], "ok": "Operator"},
{"id": 16, "stage": "lexer", "source": [55, 56], "ok": "Identifier"},
{"id": 17, "stage": "lexer", "source": [56, 57], "ok": "Operator"},
{"id": 18, "stage": "lexer", "source": [57, 58], "ok": "Operator"},
{"id": 19, "stage": "lexer", "source": [63, 69], "ok": "Keyword"},
{"id": 20, "stage": "lexer", "source": [70, 71], "ok": "Integer"},
{"id": 21, "stage": "lexer", "source": [71, 72], "ok": "Operator"},
{"id": 22, "stage": "lexer", "source": [73, 74], "ok": "Operator"},
{"id": 28, "parent_id": 26, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 26, "parent_id": 25, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 40, "parent_id": 39, "stage": "parser", "source": [26, 54], "ok": "Path"},
{"id": 49, "parent_id": 48, "stage": "parser", "source": [55, 56], "ok": "Path"},
{"id": 48, "parent_id": 46, "stage": "parser", "source": [55, 56], "ok": "Identifier"},
{"id": 46, "parent_id": 45, "stage": "parser", "source": [55, 56]},
{"id": 45, "parent_id": 44, "stage": "parser", "source": [55, 56]},
{"id": 44, "parent_id": 43, "stage": "parser", "source": [55, 56]},
{"id": 43, "parent_id": 42, "stage": "parser", "source": [55, 56]},
{"id": 42, "parent_id": 41, "stage": "parser", "source": [55, 56]},
{"id": 41, "parent_id": 39, "stage": "parser", "source": [55, 56]},
{"id": 39, "parent_id": 37, "stage": "parser", "source": [26, 57], "ok": "Routine Call"},
{"id": 37, "parent_id": 36, "stage": "parser", "source": [26, 57]},
{"id": 36, "parent_id": 35, "stage": "parser", "source": [26, 57]},
{"id": 35, "parent_id": 34, "stage": "parser", "source": [26, 57]},
{"id": 34, "parent_id": 33, "stage": "parser", "source": [26, 57]},
{"id": 33, "parent_id": 32, "stage": "parser", "source": [26, 57]},
{"id": 32, "parent_id": 29, "stage": "parser", "source": [26, 57]},
{"id": 29, "parent_id": 25, "stage": "parser", "source": [26, 58], "ok": "Statement Expression"},
{"id": 76, "parent_id": 72, "stage": "parser", "source": [70, 71], "ok": "Number"},
{"id": 72, "parent_id": 71, "stage": "parser", "source": [70, 71]},
{"id": 71, "parent_id": 70, "stage": "parser", "source": [70, 71]},
{"id": 70, "parent_id": 69, "stage": "parser", "source": [70, 71]},
{"id": 69, "parent_id": 68, "stage": "parser", "source": [70, 71]},
{"id": 68, "parent_id": 67, "stage": "parser", "source": [70, 71]},
{"id": 67, "parent_id": 66, "stage": "parser", "source": [70, 71]},
{"id": 66, "parent_id": 25, "stage": "parser", "source": [63, 71], "ok": "Return"},
{"id": 25, "parent_id": 23, "stage": "parser", "source": [0, 74], "ok": "Funtion Definition"},
{"id": 23, "stage": "parser", "source": [0, 74], "ok": "File Module"},
{"id": 79, "stage": "canonize-item-path", "source": [0, 74], "ok": "$undeclared::undeclared"},
{"id": 80, "stage": "canonize-item-path", "source": [0, 74], "ok": "$undeclared::undeclared::undeclared"},
{"id": 81, "stage": "canonize-item-path", "source": [0, 74], "ok": "$undeclared::undeclared::my_main"},
{"id": 82, "stage": "type-resolver", "source": [55, 56], "error": "L2: Could not find definition for x in this scope"}]