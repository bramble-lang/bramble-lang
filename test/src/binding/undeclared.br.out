Error: L2: Could not find definition for x in this scope
